lib/config/presets/gitlab/index.ts

```diff
@@ -52,7 +53,73 @@ export async function fetchJSONFile(
     throw new Error(PRESET_DEP_NOT_FOUND);
   }
 
-  return parsePreset(res.body, fileName);
+  // Extract the actual filename without path or query parameters
+  const cleanFileName = extractFilenameFromGitLabPath(fileName);
+
+  return parsePreset(res.body, cleanFileName);
+}
+
+/**
+ * Extracts the actual filename from a GitLab path, which could be:
+ * - A simple filename: "renovate.json5"
+ * - A path: "path/to/renovate.json5"
+ * - A URL-encoded path: "src%2Frenovate%2Fconfig.json5"
+ * - A GitLab API URL: "https://gitlab.example.com/api/v4/projects/13083/repository/files/renovate.json5?ref=main"
+ *
+ * @param fileName - The filename or path to extract from
+ * @returns The extracted filename without path or query parameters
+ */
+function extractFilenameFromGitLabPath(fileName: string): string {
+  // Fast path for simple filenames without path separators
+  if (
```

Comment:
do:
- parseUrl(filename)
- if it returns a url -> use `pathname` and remove mentioned stuff like `/raw`
- if not -> return as is. it should be a normal filename with extension